src/components/ShareTooltip/SocialShareLink/SocialShareLink.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import {stringify} from 'query-string';
+
+import {block} from '../../utils/cn';
+import {SocialShareData} from '../models';
+import {Button} from '../../Button';
+import {Icon} from '../../Icon';
+import {LayoutDirections, SocialNetwork} from '../constants';
+import {SVGIconData} from '../../Icon/types';
+import * as icons from '../../icons/social';
+
+import './SocialShareLink.scss';
+
+const b = block('social-share-link');
+
+export interface SocialShareLinkProps extends SocialShareData {
+    type: SocialNetwork;
+    icon?: SVGIconData;
+    className?: string;
+    direction?: LayoutDirections;
+}
+
+export class SocialShareLink extends React.PureComponent<SocialShareLinkProps> {
+    render() {
+        const {type, direction, icon = icons[type], className} = this.props;
+        const url = this.getShareLink(type);
+        const typeModifier = type.toLowerCase();
+
+        if (direction === 'column') {
+            return (
+                <Button
+                    view="flat"
+                    size="l"
+                    href={url}
+                    target="_blank"
+                    width="max"
+                    className={b(null, className)}
+                >
+                    <Icon data={icon} size={16} className={b('icon', {type: typeModifier})} />
+                    <span className={b(null, className)}>{SocialNetwork[type]}</span>
+                </Button>
+            );
+        }
+        return (
+            <Button view="flat" size="l" href={url} target="_blank" className={b(null, className)}>
+                <Icon data={icon} size={24} className={b('icon', {type: typeModifier})} />
+            </Button>
+        );
+    }
+
+    private getShareLink(type: SocialNetwork) {
+        const {url, title, text} = this.props;
+
+        // https://github.com/bradvin/social-share-urls
+        switch (type) {
+            case SocialNetwork.Telegram:
+                return `https://t.me/share/url?${stringify({url, text: title})}`;
```

Comment:
Can we use `URL` instead, please? It do same thing, but without external dependency

```suggestion
                const url = new URL('https://t.me/share/url');
                url.searchParams.set('url', url);
                url.searchParams.set('text', title);
                return url.toString();
```